test/functional/collection_tests.js

```diff
@@ -79,1581 +73,1010 @@ describe('Collection', function() {
+      db.createCollection('shouldFailDueToExistingCollection', { strict: true }, (err, coll) => {
+        expect(err).to.not.exist;
+        expect(coll).to.exist;
 
-  /**
-   * @ignore
-   */
-  it('should fail due to exiting collection', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
+        db.createCollection('shouldFailDueToExistingCollection', { strict: true }, err => {
+          expect(err).to.be.an.instanceof(Error);
+          expect(err.message).to.equal(
+            'Collection shouldFailDueToExistingCollection already exists. Currently in strict mode.'
+          );
+          client.close(done);
+        });
       });
+    });
 
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
-
-        db.createCollection('shouldFailDueToExistingCollection', { strict: true }, function(
-          err,
-          coll
-        ) {
-          test.equal(null, err);
-          test.ok(coll != null);
-
-          db.createCollection('shouldFailDueToExistingCollection', { strict: true }, function(err) {
-            test.ok(err != null);
+    /**
+     * @ignore
+     */
+    it('should filter correctly during list', function(done) {
+      const testCollection = 'integration_tests_collection_123'; // The collection happens to contain the database name
+      // Create a collection
+      db.createCollection(testCollection, err => {
+        expect(err).to.not.exist;
 
-            client.close();
-            done();
+        db.listCollections({ name: testCollection }).toArray((err, documents) => {
+          expect(err).to.not.exist;
+          expect(documents.length).to.equal(1);
+          let found = false;
+          documents.forEach(document => {
+            if (document.name === testCollection) found = true;
           });
+          expect(found).to.be.true;
+          client.close(done);
         });
       });
-    }
-  });
+    });
 
-  /**
-   * @ignore
-   */
-  it('should filter correctly during list', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var testCollection = 'integration_tests_collection_123'; // The collection happens to contain the database name
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
-      });
+    /**
+     * @ignore
+     */
+    it('should filter correctly with index during list', function(done) {
+      const testCollection = 'collection_124';
+      // Create a collection
+      db.createCollection(testCollection, err => {
+        expect(e
```

Comment:
Currently failing for version 2.6

Reply:
I think we can get rid of this `names` array and instead just check that each `collection.name` in the `collections` array is what we expect it to be.